tests/unit_tests/test_metagenome.py

```diff
@@ -1,89 +1,92 @@
+    assert mg.largest_seq == "NODE_1_length_1389215_cov_225.275"
     assert mg.nseqs == 2059
     assert mg.size == 78034179
-    with pytest.raises(FileExistsError):
-        test_metagenome.length_filter(out=out, cutoff=3000)
+    assert round(mg.length_weighted_gc, 2) == 54.39
 
+    with pytest.raises(ValueError):
+        init_Metagenome.length_filter(out=out_fpath, cutoff="invalid_cutoff")
+    with pytest.raises(ValueError):
+        init_Metagenome.length_filter(out=out_fpath, cutoff=-6)
+    with pytest.raises(FileExistsError):
+        init_Metagenome.length_filter(out=out_fpath, cutoff="3000", force=False)
 
-def test_call_orfs(test_metagenome):
-    # TODO: Add mock here for prodigal call
-    assert test_metagenome.orfs_called is False
-    nucls_fp, prots_fp = test_metagenome.call_orfs(cpus=4, parallel=True)
-    assert nucls_fp == test_metagenome.nucl_orfs_fpath
-    assert prots_fp == test_metagenome.prot_orfs_fpath
-    assert test_metagenome.orfs_called
-    assert len(test_metagenome.prots) == 75390
-    assert len(test_metagenome.nucls) == 75390
 
+@patch(
+    "autometa.common.external.prodigal.run",
+    return_value=("nucls_out_fpath", "prots_out_fpath"),
+)
+def test_call_orfs(patched_progial_run, init_Metagenome):
+    with pytest.raises(TypeError):
+        init_Metagenome.call_orfs(force="invalid_input")
+    with pytest.raises(TypeError):
+        init_Metagenome.call_orfs(parallel="invalid_input")
+    with pytest.raises(TypeError):
+        init_Metagenome.call_orfs(cpus="invalid_input")
 
-def test_get_kmers(test_metagenome):
-    out = os.path.join(test_metagenome.outdir, "kmers.tsv")
-    kmers_df = test_metagenome.get_kmers(
-        kmer_size=5, multiprocess=True, out=out, nproc=4
-    )
-    assert kmers_df.shape == (3587, 512)
-    normalized = os.path.join(test_metagenome.outdir, "kmers.normalized.tsv")
-    normalized_df = test_metagenome.get_kmers(
-        kmer_size=5,
-        multiprocess=True,
-        out=out,
-        nproc=4,
-        normalized=normalized,
-        force=True,
-    )
-    assert normalized_df.shape == (3587, 512)
-
-
-def test_get_coverages(test_metagenome):
-    out = os.path.join(test_metagenome.outdir, "spades_coverages.tsv")
-    cov_df = test_metagenome.get_coverages(out=out, from_spades=True)
-    assert isinstance(cov_df, pd.DataFrame)
-    assert cov_df.shape == (3587, 1)
-    assert "coverage" in cov_df.columns
-
-
-def test_assign_taxonomy(test_metagenome):
-    assert test_metagenome.taxonomy_assigned is False
-    params = {
-        "cpus": 1,
-        "ncbi": "</path/to/mocked/ncbi/dir>",
-        "verbose": False,
-        "blast": "</path/to/mocked/blastp.tsv",
-        "hits": "</path/to/mocked/hits.pkl.gz>",
-        "force": True,
-    }
-    # taxa_df = test_metagenome.assign_taxonomy(force=False, kwargs=params)
-    # assert taxa_df.shape == (3587, )
+    nucls_fp, prots_fp = init_Metagenome.call_orfs()
```

Comment:
`self.call_orfs()` has three parameters that should be tested here.

### 😍 🤖 💚  Loving the use of patching prodigal 👍 

Would we be able to do introspection here to ensure the `prodigal.run(...)` command looks as what we would expect when we call `self.call_orfs(...)`?

#### Args
* force
* cpus
* parallel

i.e.
```python
with pytest.raises(TypeError):
    nucls_fp, prots_fp = init_Metagenome.call_orfs(force="false")

with pytest.raises(TypeError):
    nucls_fp, prots_fp = init_Metagenome.call_orfs(parallel="F")

with pytest.raises(TypeError):
    nucls_fp, prots_fp = init_Metagenome.call_orfs(cpus=4.0)
```

### What about when the ORFs files already exist?
We also need to ensure the `self.prot_orfs_fpath` and `self.nucl_orfs_fpath` are returned when the ORFs files already exist.